scrapers/dc_onlyfans_fansdb/dc_onlyfans_fansdb.py

```diff
@@ -402,78 +415,84 @@ def format_title(title, username, date, scene_index):
     scene_info = f' ({scene_index})' if scene_index > 0 else ''
     return f'{f_title}{scene_info}'
 
-def process_row(row, username, scene_index=0):
+
+def process_row(row, username, network, scene_index=0):
     """
     Process a database row and format post details.
     """
-    date = row[2].strftime("%Y-%m-%d")
-    title = format_title(row[1], username, date, scene_index)
-    details = row[1]
-    code = str(row[0])
-    return {"title": title, "details": details, "date": date, "code": code}
-
-def find_metadata_db(start_path, username):
+    res = {}
+    res['date'] = row[2].strftime("%Y-%m-%d")
+    res['title'] = format_title(row[1], username, res['date'], scene_index)
+    res['details'] = row[1]
+    res['code'] = str(row[0])
+    if network == 'OnlyFans':
+        res['urls'] = [f"https://onlyfans.com/{res['code']}/{username}"]
+    elif network == 'Fansly':
+        res['urls'] = [f"https://fansly.com/post/{res['code']}"]
+    return res
+
+
+def get_metadata_db(search_path, username, network):
     """
-    Recursively search for 'user_data.db' file upwards starting from 'start_path'.
+    Recursively search for 'user_data.db' file starting from 'search_path'
     """
-    start_path = Path(start_path).resolve()
+    search_path = Path(search_path).resolve()
 
-    while start_path != start_path.parent:
-        # Search for user_data.db in the current directory and its subdirectories
-        for db_file in start_path.rglob(username + '/**/user_data.db'):
-            if db_file.is_file():
-                return db_file
+    while search_path != search_path.parent:
+        db_files = list(search_path.rglob(f"{network}/**/{username}/**/user_data.db"))
+        db_files = [db for db in db_files if db.is_file()]
+        if db_files:
+            return db_files[0]
 
-        start_path = start_path.parent
+        search_path = search_path.parent
 
     return None
 
-def username_from_path(path, after='OnlyFans'):
+
+def get_path_info(path):
     """
-    Extract the username from a given path
+    Extract the username and network from a given path
     """
-    path = path.parts
+    network = 'Fansly' if 'Fansly' in str(path) else 'OnlyFans'
+    parts = path.parts
     try:
-        index = path.index(after)
-        if index + 1 < len(path):
-            return path[index + 1]
-        else:
-            return None  # Element found but no item follows it
+        index = parts.index(network)
+        if index + 1 < len(parts):
+            return parts[index + 1], network, Path(*parts[:index + 2])
+
+        raise ValueError
     except ValueError:
```

Comment:
There really is no point in raising a ValueError if you're going to catch it on the very next line 😁 Just drop these two lines